cmd/kubeadm/app/cmd/phases/join/kubelet.go

```diff
@@ -217,6 +220,21 @@ func runKubeletStartJoinPhase(c workflow.RunData) (returnErr error) {
 		return err
 	}
 
+	if cfg.ControlPlane != nil && features.Enabled(initCfg.FeatureGates, features.WaitForAllControlPlaneComponents) {
+		fmt.Println("[kubelet-start] Wait for control plane components")
+		timeout := 40 * time.Second
```

Comment:
and following on the prev comment, there should be possible to use `clusterConfig.APIServer.TimeoutForControlPlane.Duration` instead of 40s.

FTR, we have this pending:
https://github.com/kubernetes/kubernetes/pull/122529

but my proposal is to still have a single timeout value for any CP component - 4m.